addons/website_payment/tests/test_snippets.py

```diff
@@ -8,5 +11,12 @@ class TestSnippets(odoo.tests.HttpCase):
     def test_01_donation(self):
         demo_provider = self.env['payment.provider'].search([('code', '=', "demo")])
         demo_provider.write({'state': 'test'})
+        payment_demo = self.env['ir.module.module']._get('payment_demo')
+
+        if payment_demo.state != 'installed':
+            self.skipTest("payment_demo module is not installed")
```

Comment:
This should be checked before searching for the Demo provider and trying to enable it.